File: hw06-jpa/src/main/java/ru/otus/hw/models/Book.java
package ru.otus.hw.models;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Id;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.NamedEntityGraph;
import jakarta.persistence.NamedAttributeNode;
import jakarta.persistence.OneToOne;
import jakarta.persistence.CascadeType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.NamedAttributeNode;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "books")
@NamedEntityGraph(name = "book-entity-graph",
        attributeNodes = { @NamedAttributeNode("author"), @NamedAttributeNode("genre") })
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String title;

    @OneToOne(targetEntity = Author.class, cascade = CascadeType.PERSIST)
    @JoinColumn(name = "author_id")
    private Author author;

    @OneToOne(targetEntity = Genre.class, cascade = CascadeType.PERSIST)
    @JoinColumn(name = "genre_id")
    private Genre genre;

}

File: hw10-singlepage_back/src/main/ui-webapp/src/app/model/book.ts
import {Author} from "./author";
import {Genre} from "./genre";

export class Book {
  id?: number | undefined;
  title?: string | null;
  author?: Author | null | undefined;
  genre?: Genre | null | undefined;

  constructor() {}

  toUpdateCreateJson() {
    return {
      "id": this.id,
      "title": this.title,
      "authorId": (this.author === null || this.author === undefined) ? null : this.author.id,
      "genreId": (this.genre === null || this.genre === undefined) ? null : this.genre.id
    }
  }
}
